.github/workflows/deploy_terraform.yml

```diff
@@ -107,26 +120,25 @@ jobs:
         working-directory: operations/app/terraform/vars/${{ needs.pre_job.outputs.env_name }}
     steps:
       - name: Check Out Changes
-        uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
+        uses: actions/checkout@v4
       - name: Connect to VPN and login to Azure
         uses: ./.github/actions/vpn-azure
         with:
           env-name: ${{ needs.pre_job.outputs.env_name }}
           tls-key: ${{ secrets.TLS_KEY }}
-          ca-cert: ${{ secrets.CA_CRT}}
+          ca-cert: ${{ secrets.CA_CRT }}
           user-crt: ${{ secrets.USER_CRT }}
           user-key: ${{ secrets.USER_KEY }}
           sp-creds: ${{ env.AZURE_CREDENTIALS }}
           tf-auth: true
-      - name: Use specific version of Terraform
-        uses: hashicorp/setup-terraform@b9cd54a3c349d3f38e8881555d616ced269862dd
-        with:
-          terraform_version: 1.7.4
-          terraform_wrapper: false
-      - name: Run Terraform
+      - name: Run Terraform Sanity Checks
         run: |
-          terraform init -input=false
+          terraform workspace select -or-create ${{ needs.pre_job.outputs.env_name }}
```

Comment:
the first thing wwe need to run before workspace selection is 
`terraform init`